src/main/webapp/app/complaints/complaints-for-students/complaints-student-view.component.ts

```diff
@@ -148,14 +151,15 @@ export class ComplaintsStudentViewComponent implements OnInit {
         if (!actionThresholdInDays) {
             return false;
         }
-        const isWithinThreshold = dayjs().isSameOrBefore(dayjs(completionDate).add(actionThresholdInDays, 'day'));
-        if (!this.exercise.assessmentDueDate) {
-            return isWithinThreshold;
-        }
-        if (dayjs().isAfter(dayjs(this.exercise.assessmentDueDate))) {
-            return isWithinThreshold || dayjs().isSameOrBefore(dayjs(this.exercise.assessmentDueDate).add(actionThresholdInDays, 'day'));
+        let startOfComplaintTime = this.result?.completionDate;
+        startOfComplaintTime =
+            !startOfComplaintTime || (this.exercise.dueDate && this.exercise.dueDate.isAfter(startOfComplaintTime)) ? this.exercise.dueDate : startOfComplaintTime;
+        startOfComplaintTime =
+            this.exercise.assessmentDueDate && this.exercise.assessmentDueDate.isAfter(startOfComplaintTime) ? this.exercise.assessmentDueDate : startOfComplaintTime;
+        if (!startOfComplaintTime) {
+            return false;
         }
```

Comment:
```suggestion
        if (!this.result?.completionDate) {
            return false;
        }
        let startOfComplaintTime = this.result!.completionDate;
        if (this.exercise.dueDate && this.exercise.dueDate.isAfter(startOfComplaintTime) {
            startOfComplaintTime = this.exercise.dueDate;
        }
        if (this.exercise.assessmentDueDate && this.exercise.assessmentDueDate.isAfter(startOfComplaintTime) {
            startOfComplaintTime = this.exercise.assessmentDueDate;
        }
```

```suggestion
        if (!this.result?.completionDate) {
            return false;
        }
        const possibleStartDates = [this.result.completionDate, this.exercise.dueDate, this.exercise.assessmentDate];
        const startOfComplaintTime = dayjs.max(possibleStartDates.filter((date) => date !== undefined) as Dayjs[]);
```

I would suggest one of those two implementations instead to make it more intuitive to understand.
2) is way shorter but requires the filter and casting, 1) is more classic style

I would prefer 2 though imo. Short and neat.